### PR TITLE
nrf_modem: doc: update NRF_SO_RCVTIMEO for accept

### PR DESCRIPTION
**Pull request by eivindj-nordic**
The NRF_SO_RCVTIMEO is used to set the timeout in accept.

### CODE REVIEW

File: nrf_modem/doc/sockets/socket_options_func.rst [hunk 1]
```diff
@@ -111,10 +111,10 @@ NRF_SO_REUSEADDR
   The default value for :c:macro:`NRF_SO_REUSEADDR` is ``off``, that is, reuse of local addresses is not permitted.
 
 NRF_SO_RCVTIMEO
-   Set a timeout value for the :c:func:`nrf_recv` and :c:func:`nrf_recvfrom` operations.
+   Set a timeout value for the :c:func:`nrf_recv`, :c:func:`nrf_recvfrom` and :c:func:`nrf_accept` operations.
    This option accepts an :c:struct:`nrf_timeval` structure with a number of seconds and microseconds specifying the limit on how long to wait for an input operation to complete.
-   If a receive operation has blocked for this much time without receiving additional data, it returns with a partial count, or ``errno`` is set to ``NRF_EAGAIN`` or ``NRF_EWOULDBLOCK`` if no data were received.
-   The default for this option is the value ``0``, which indicates that a receive operation will not time out.
+   If a receive or accept operation has blocked for this much time without receiving additional data, it returns with a partial count, or ``errno`` is set to ``NRF_EAGAIN`` or ``NRF_EWOULDBLOCK`` if no data were received.
```
Comment (lemrey) [inline nrf_modem/doc/sockets/socket_options_func.rst:116]: maybe "input" operation?

Comment (anhmolt) [inline nrf_modem/doc/sockets/socket_options_func.rst:116]: Will `nrf_accept()` ever return with a "partial count"? The `nrf_accept()` seems to return with either `-1` or file descriptor.

I see this sentence originates from [here](https://pubs.opengroup.org/onlinepubs/9699919799/functions/V2_chap02.html#tag_15_10_16:~:text=If%20a%20receive%20operation%20has%20blocked%20for%20this%20much%20time%20without%20receiving%20additional%20data%2C%20it%20shall%20return%20with%20a%20partial%20count%20or%20errno%20shall%20be%20set%20to%20%5BEAGAIN%5D%20or%20%5BEWOULDBLOCK%5D%20if%20no%20data%20were%20received.).

Comment (lemrey) [reply nrf_modem/doc/sockets/socket_options_func.rst:116]: Good point, maybe it's best to split the sentence in two, one for receive operations and another for accept operations?

Comment (eivindj-nordic) [reply nrf_modem/doc/sockets/socket_options_func.rst:116]: Will split it in two sentences.

File: nrf_modem/doc/sockets/socket_options_func.rst [hunk 2]
```diff
@@ -111,10 +111,11 @@ NRF_SO_REUSEADDR
   The default value for :c:macro:`NRF_SO_REUSEADDR` is ``off``, that is, reuse of local addresses is not permitted.
 
 NRF_SO_RCVTIMEO
-   Set a timeout value for the :c:func:`nrf_recv` and :c:func:`nrf_recvfrom` operations.
+   Set a timeout value for the :c:func:`nrf_recv`, :c:func:`nrf_recvfrom` and :c:func:`nrf_accept` operations.
    This option accepts an :c:struct:`nrf_timeval` structure with a number of seconds and microseconds specifying the limit on how long to wait for an input operation to complete.
    If a receive operation has blocked for this much time without receiving additional data, it returns with a partial count, or ``errno`` is set to ``NRF_EAGAIN`` or ``NRF_EWOULDBLOCK`` if no data were received.
-   The default for this option is the value ``0``, which indicates that a receive operation will not time out.
+   If an accept operation has blocked for this much time without receiving an incomming connection, it returns an error and ``errno`` is set to ``NRF_EAGAIN``.
```
Comment (lemrey) [inline nrf_modem/doc/sockets/socket_options_func.rst:117]: `incoming`

**Review by lemrey — COMMENTED** [review]
Use `:c:macro:` for errnos

**Comment by lemrey** [comment]
It looks good, if you could keep a separate commit even better so it's easier to review.

**Comment by anhmolt** [comment]
Found some more things we can improve while reviewing this, so I opened https://github.com/lemrey/sdk-nrfxlib/pull/42.

**Review by anhmolt — APPROVED** [review]

File: nrf_modem/doc/fault_handling.rst [hunk 3]
```diff
@@ -35,19 +35,19 @@ The application can reinitialize the modem by reinitializing the Modem library t
 .. important::
    If modem traces are enabled, the modem continues to output trace data (a coredump) in the event of a fault, even after signaling the fault to the application.
    To make sure the application is able to retrieve the coredump correctly, the application must not re-initialize the modem until all the outstanding trace data has been processed.
-   The :c:func:`nrf_modem_trace_get()` function will return ``-ENODATA`` when all outstanding trace data has been processed by the application.
+   The :c:func:`nrf_modem_trace_get()` function will return :c:macro:`-NRF_ENODATA` when all outstanding trace data has been processed by the application.
```
Comment (lemrey) [inline nrf_modem/doc/fault_handling.rst:38]: Does it work to have the `-` minus in the macro name? Those generate links in HTML to the macro definition, how are they working with the minus sign in front?

Comment (anhmolt) [reply nrf_modem/doc/fault_handling.rst:38]: That is a good point..
Maybe do as @MirkoCovizzi suggested in #42, and write
```
return negative :c:macro:`NRF_ENODATA`
```

Comment (anhmolt) [reply nrf_modem/doc/fault_handling.rst]: Last changes ended up in the wrong commit, if I'm not mistaken.

**Review by lemrey — APPROVED** [review]

**Pull request merged**